crates/cxx-qt-gen/src/generator/rust/mod.rs

```diff
@@ -18,5 +25,120 @@ pub struct GeneratedRustBlocks {
     /// Ident of the namespace of the QObject
     pub namespace: String,
     /// Generated QObject blocks
-    pub qobjects: Vec<qobject::GeneratedRustQObject>,
+    pub qobjects: Vec<GeneratedRustQObject>,
+}
+
+impl GeneratedRustBlocks {
+    pub fn from(parser: &Parser) -> Result<GeneratedRustBlocks> {
+        Ok(GeneratedRustBlocks {
+            cxx_mod: parser.passthrough_module.clone(),
+            cxx_mod_contents: vec![generate_include(parser)?],
+            cxx_qt_mod_contents: parser.cxx_qt_data.uses.clone(),
+            namespace: parser.cxx_qt_data.namespace.clone(),
+            qobjects: parser
+                .cxx_qt_data
+                .qobjects
+                .values()
+                .map(GeneratedRustQObject::from)
+                .collect::<Result<Vec<GeneratedRustQObject>>>()?,
+        })
+    }
+}
+
+/// Generate the include line for this parsed block
+fn generate_include(parser: &Parser) -> Result<Item> {
+    // TODO: for now the cxx stem comes from the first QObject name
+    // but later this may come from the module or file so would then have
+    // a field on the cxx_qt_data
+    if parser.cxx_qt_data.qobjects.len() != 1 {
+        return Err(Error::new(
+            parser.passthrough_module.span(),
+            "Only one QObject is currently supported in the ItemMod.",
+        ));
+    }
+    let (qt_ident, _) = parser.cxx_qt_data.qobjects.iter().take(1).next().unwrap();
+    let import_path = format!(
+        "cxx-qt-gen/include/{}.cxxqt.h",
+        cxx_stem_from_ident(qt_ident)
+    );
+
+    syn::parse2(quote! {
+        unsafe extern "C++" {
+            include!(#import_path);
+        }
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::tests::{assert_tokens_eq, tokens_to_syn};
+
+    #[test]
+    fn test_generated_rust_blocks() {
+        let module: ItemMod = tokens_to_syn(quote! {
+            #[cxx_qt::bridge]
+            mod ffi {
+                #[cxx_qt::qobject]
+                struct MyObject;
+            }
+        });
+        let parser = Parser::from(module).unwrap();
+
+        let rust = GeneratedRustBlocks::from(&parser).unwrap();
+        assert_eq!(rust.cxx_mod.content.unwrap().1.len(), 0);
+        assert_eq!(rust.cxx_mod_contents.len(), 1);
```

Comment:
So `cxx_mod` is the module (like `mod ffi { blah }` and `cxx_mod_contents` is a vector of contents that are added to the module later.

Not sure how best to name them as passthrough is something different and not this as cxx_mod_contents are generated.

Note that ItemMod has a field call `content` hence where `contents` has come from.

Reply:
It could inject it in, it does cause some code duplication, and then changes the ordering of components a bit - I can't remember if this was the reason it doesn't work as it then affects the include order.... let me try it again...

Reply:
In `write_rust`, there is:
```rust
    // Inject the CXX blocks
    if let Some((_, items)) = &mut cxx_mod.content {
        items.extend(cxx_mod_contents.into_iter());
    } else {
        cxx_mod.content = Some((syn::token::Brace::default(), cxx_mod_contents));
    }
```

Could that be moved to `GeneratedRustBlocks::from` to get rid of the confusingly similar fields?

Reply:
That block needs to stay there are the write phase also adds blocks in.

Copying that block into the `GeneratedRustBlocks::from` allows removal of the `cxx_mod_contents`, but this code would still need to be in `write_rust` as well as it also needs to inject blocks into the `cxx_mod`.

Reply:
This was originally why i split it out into a vec that is passed along, as then there is only one place that needs to inject the blocks into `cxx_mod`. As both the generator and write phase have global blocks to add.

Reply:
okay